src/main/java/net/minecraftforge/fluids/FluidUtil.java

```diff
@@ -143,7 +143,9 @@ public static FluidActionResult tryFillContainer(@NotNull ItemStack container, I
                         }
                         else
                         {
-                            containerFluidHandler.fill(simulatedTransfer, IFluidHandler.FluidAction.SIMULATE);
+                            //The result should be the same regardless of weather we are simulating or not.
+                            //So we must execute the transfer on the copied item stack so the resultContainer is correct.
```

Comment:
I think this is mostly correct (barring any handlers that use the stack as a reference handler such as for changing something in the world, but there isn't a great way to handle those), but I think it would make sense to have this comment be basically identical to the one below in `tryEmptyContainer` as its first sentence is better at explaining why it is fine to actually execute even though it is is in the do not fill logic branch.